### PR TITLE
ethtool: use ioctl wrapper from golang.org/x/sys/unix

### PR DESCRIPTION
**Pull request by tklauser**
Update the vendored copy of `golang.org/x/sys/unix` to pull inioctl wrapper fixes and switch `pkg/datapath/ethtool` to use one of these wrappers with the correctly padded `ifreqData struct`.

See individual commits for details.

### CODE REVIEW

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-me-please

File: pkg/datapath/linux/ethtool/ethtool_linux.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	if err != nil {
+		return false, err
+	}
+	defer unix.Close(fd)
+
+	info, err := unix.IoctlGetEthtoolDrvinfo(fd, iface)
+	if err != nil {
+		return false, err
+	}
+	return string(bytes.TrimRight(info.Driver[:], "\x00")) == "veth", nil
```
Comment (ti-mo) [inline pkg/datapath/linux/ethtool/ethtool_linux.go:23]: Is this only ever called on veth ifs? Shouldn't this include bridges and mac/ipvlan as well? If not, the fn name might be a bit misleading.

```
# ethtool -i eth1
driver: macvlan
...
```

Comment (tklauser) [reply pkg/datapath/linux/ethtool/ethtool_linux.go:23]: Currently, the only call site is here:

https://github.com/cilium/cilium/blob/2d075ea34720ed625e297ede67ace8dce5bbdb53/pkg/datapath/loader/base.go#L182

used in ENI IPAM mode top determine which interfaces to use for IPSec encryption. I'm not very familiar with that, so I'd have to defer to @jrfastab to answer the question of whether we should include bridges and mac/ipvlan as well.

**Review by rolinh — APPROVED** [review]

**Comment by tklauser** [comment]
Rebased to pick up fixes for EKS CI 3.0 tests.

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-gke

**Review by ti-mo — APPROVED** [review]

**Comment by tklauser** [comment]
test-1.16-netnext

**Comment by tklauser** [comment]
Rebased and dropped unrelated file renames which I accidentally committed.

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-1.16-netnext

**Comment by tklauser** [comment]
test-1.16-netnext

**Comment by jrfastab** [comment]
Is the incorrect padding causing an issue now? Meaning should this be marked as a bugfix?

**Review by jrfastab — APPROVED** [review]

**Comment by tklauser** [comment]
> Is the incorrect padding causing an issue now? Meaning should this be marked as a bugfix?

Good point. I'm currently not aware of any issues this is causing, but I guess it could potentially cause issues. These will likely be hard to track down. I'll thus mark it as bugfix and for backports.

**Comment by tklauser** [comment]
test-me-please

**Pull request merged**